truss/templates/server/common/truss_server.py

```diff
@@ -1,248 +1,169 @@
+from kserve.errors import InvalidInput
+from kserve.handlers import DataPlane, ModelRepositoryExtension, V1Endpoints
 from model_wrapper import ModelWrapper
-from pythonjsonlogger import jsonlogger  # noqa: E402
-from tornado.httpserver import HTTPServer
-from tornado.ioloop import IOLoop
-
-ensure_kfserving_installed()
+from starlette.responses import Response
 
 logger = logging.getLogger(__name__)
 
 
-def _configure_logging():
-    json_log_handler = logging.StreamHandler()
-    json_log_handler.setFormatter(
-        jsonlogger.JsonFormatter("%(asctime)s %(levelname)s %(message)s")
-    )
-    logger = logging.getLogger()
-    for handler in logger.handlers:
-        logger.removeHandler(handler)
-    logger.addHandler(json_log_handler)
+async def parse_body(request: Request):
+    data: bytes = await request.body()
+    return data
+
+
+class BasetenEndpoints(V1Endpoints):
+    def __init__(
+        self,
+        dataplane: DataPlane,
+        model_repository_extension: Optional[ModelRepositoryExtension] = None,
+    ):
+        super().__init__(dataplane, model_repository_extension)
+
+    @staticmethod
+    def check_healthy(model: ModelWrapper):
+        model.start_load()
+
+        if model.load_failed():
+            raise errors.InferenceError("Model load failed")
+
+        if not model.ready:
+            raise errors.ModelNotReady(model.name)
 
+    async def model_ready(self, model_name: str) -> Dict[str, Union[str, bool]]:
+        model: ModelWrapper = self.dataplane.get_model_from_registry(model_name)
+
+        self.check_healthy(model)
+
+        return {}
+
+    def predict(
+        self, model_name: str, request: Request, body_raw: bytes = Depends(parse_body)
+    ) -> Response:
+        model: ModelWrapper = self.dataplane.get_model_from_registry(model_name)
+
+        self.check_healthy(model)
+
+        body: dict
+        if self.is_binary(request):
+            body = truss_msgpack_deserialize(body_raw)
+        else:
+            body = json.loads(body_raw)
+
+        body = assign_request_to_inputs_instances_after_validation(body)
 
-class TrussHTTPBinaryHandler(HTTPHandler):
-    def validate(self, request):
         if (
             "instances" in request
             and not isinstance(request["instances"], (list, np.ndarray))
-        ) or (
-            "inputs" in request
+            or "inputs" in request
             and not isinstance(request["inputs"], (list, np.ndarray))
         ):
-            raise tornado.web.HTTPError(
-                status_code=HTTPStatus.BAD_REQUEST,
-                reason='Expected "instances" or "inputs" to be a list or NumPy ndarray',
+            raise InvalidInput(
+                'Expected "instances" or "inputs" to be a list or NumPy ndarray'
             )
-        return assign_request_to_inputs_instances_after_validation(request)
 
+        response = model.predict(body, headers=dict(request.headers.items()))
```

Comment:
I was surprised to see headers being passed here, maybe leave a comment in code to explain.

Reply:
This is part of the contract for `kserve.Model`.  I believe it was for the previous version, we were just never passing them in.  We discard them when we send them to the underlying model.

Reply:
gotcha, makes sense.